ci/scripts/utils/rocotostat.py

```diff
@@ -39,65 +55,138 @@ def input_args():
+
+    rocoto_status = {}
+    status_counts = Counter(case for sublist in rocotostat_output for case in sublist)
     for case in status_cases:
-        rocoto_status[case] = sum([sublist.count(case) for sublist in rocotostat_output_all])
+        rocoto_status[case] = status_counts[case]
 
     return rocoto_status
 
 
+def is_done(rocoto_status):
+    """
+    is_done Check if all cycles are done.
+
+    is_done(rocoto_status) checks if the total number of cycles equals the number of
+    done cycles in the rocoto_status dictionary.
+
+    Input:
+    rocoto_status - A dictionary with the count of each status case.
+
+    Output:
+    boolean - True if all cycles are done, False otherwise.
+    """
+
+    if rocoto_status['CYCLES_TOTAL'] == rocoto_status['CYCLES_DONE']:
+        return True
+    else:
+        return False
+
+
+def is_stalled(rocoto_status):
+    """
+    is_stalled Check if all cycles are stalled.
+
+    is_stalled(rocoto_status) checks if all cycles are stalled by verifying if
+    there are no jobs that are RUNNING, SUBMITTING, or QUEUED.
+
+    Input:
+    rocoto_status - A dictionary with the count of each status case.
+
+    Output:
+    boolean - True if all cycles are stalled, False otherwise.
+    """
+
+    if rocoto_status['RUNNING'] + rocoto_status['SUBMITTING'] + rocoto_status['QUEUED'] == 0:
+        return True
+    else:
+        return False
+
+
 if __name__ == '__main__':
+    """
+    main Execute the script.
+
+    main() parses the input arguments, checks if the rocotostat command is available,
+    adds default arguments to the rocotostat command, and runs it and reports
+    out to stdout spcific information of rocoto workflow.
+    """
 
     args = input_args()
 
-    error_return = 0
-    rocoto_status = rocoto_statcount()
+    try:
+        rocotostat = which("rocotostat")
+    except CommandNotFoundError:
+        logger.exception("rocotostat not found in PATH")
+        raise CommandNotFoundError("rocotostat not found in PATH")
+
+    rocotostat.add_default_arg(['-w', os.path.abspath(args.w.name), '-d', os.path.abspath(args.d.name)])
 
-    if rocoto_status['CYCLES_TOTAL'] == rocoto_status['CYCLES_DONE']:
+    rocoto_status = rocoto_statcount(rocotostat)
+    rocoto_status.update(rocotostat_summary(rocotostat))
+
+    if is_done(rocoto_status):
         rocoto_state = 'DONE'
     elif rocoto_status['DEAD'] > 0:
         error_return = rocoto_status['FAIL'] + rocoto_status['DEAD']
         rocoto_state = 'FAIL'
     elif 'UNKNOWN' in rocoto_status:
         error_return = rocoto_status['UNKNOWN']
         rocoto_state = 'UNKNOWN'
-    elif rocoto_status['RUNNING'] + rocoto_status['SUBMITTING'] + rocoto_status['QUEUED'] == 0:
-        #
-        #  TODO for now a STALLED state will be just a warning as it can
-        #  produce a false negative if there is a timestamp on a file dependency.
-        #
-        #   error_return = -3
+    elif attempt_multiple_times(is_stalled(rocoto_status), 3, 90):
```

Comment:
`is_stalled()` isn't making any system calls, so I don't think it needs `attempt_multiple_times` (it was used when rocoto_status was determined).
```suggestion
    elif is_stalled(rocoto_status):
```

Reply:
Yes I saw that and have made this change already, guess I didn't check it in yet.

Reply:
@WalterKolczynski-NOAA Oh wait, yes this is the "old" code.  Updating it now with what is in develop now.